t/helper/test-hashmap.c

```diff
@@ -243,7 +243,7 @@ int cmd__hashmap(int argc, const char **argv)
 
 			/* test that strintern works */
 			const char *i1 = strintern(p1);
-			const char *i2 = strintern(p1);
+			const char *i2 = strintern(p2);
```

Comment:
The intention of this test is to verify that `strintern()` returns the same interned string, no matter how often you call it on an identical string.

To do that, it uses `p1` twice. Your change breaks that. So this hunk is actually (almost obviously) incorrect.

(The condition `if (!strcmp("intern", cmd) && p1)` that does explicitly *not* check for `&& p2` is another clue.)